pylint/config/argument.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Definition of an Argument class and validators for various argument types.
+
+An Argument instance represents a pylint option to be handled by an argparse.ArgumentParser
+"""
+
+
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from pylint import utils as pylint_utils
+
+_ArgumentTypes = Union[str, List[str]]
+"""List of possible argument types."""
+
+
+def _csv_validator(value: Union[str, List[str]]) -> List[str]:
+    """Validates a comma separated string."""
+    return pylint_utils._check_csv(value)
+
+
+_ASSIGNMENT_VALIDATORS: Dict[str, Callable[[Any], _ArgumentTypes]] = {
+    "choice": str,
+    "csv": _csv_validator,
+}
+"""Validators for all assignment types."""
+
+
+class _Argument:
+    """Class representing an argument to be passed by an argparse.ArgumentsParser.
+
+    This is based on the parameters passed to argparse.ArgumentsParser.add_message.
+    See:
+    https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument
+    """
+
+    def __init__(
+        self,
+        name: List[str],
```

Comment:
Maybe `flags`? or `names`? It's odd for `name` to be a list. (And most often a list of one name...)

Reply:
The documentation refers to it as `name or flags`. I will change it to flags.